tests/micro/arduino/test_arduino_error_detection.py

```diff
@@ -0,0 +1,51 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import pathlib
+import re
+import sys
+
+import pytest
+
+import conftest
+from tvm.micro.project_api.server import ServerError
+
+
+@pytest.fixture(scope="function")
+def workspace_dir(request, platform):
+    return conftest.make_workspace_dir("arduino_error_detection", platform)
```

Comment:
Given these fixtures are only used as part of the tests in this folder, and `conftest` is meant for auto-registration or pytest configuration rather than to hold helpers and utils, could these be either moved fully into `conftest` as fixtures or the underlying function moved into a more suitably named file?

Reply:
The scope fixture of `workspace_dir` varies from test to test (for example, test_arduino_workflow` has it at module scope), and I suspect tests with more complex `workspace_dir` setups will be added in the future. For that reason, I'd prefer to define `workspace_dir` individually in each test, but use the helper function `make_workspace_dir` to avoid code reuse. I've also added a comment to each of these functions to clarify it.

That said, if you feel strongly I'm also open to other solutions, like defining multiple kinds of workspace dirs in `conftest` (e.g. a `module_scope_workspace_dir`, `per_function_workspace_dir`, etc.).

Reply:
Argh, I feel like I could've been clearer here, my apologies. My suggestion is not to have general functions in `conftest.py` but rather in a different file such as a `workspace.py` to include from the test. This means only pytest-specific things end up in `conftest.py`, I have no issue with fixtures local to the test files which then use those functions.

Reply:
Ah, that makes more sense. I like the idea of moving non-pytest code into a `workspace.py`, but I think that might be OOS for this PR. Definitely worth addressing in a follow up though!

Reply:
Agreed, it'd be good to have the move isolated in another PR :+1:

Reply:
i agree with @Mousius sentiment--let's place test_utils in a file called test_utils or similar. can be a follow-up PR.